### ISSUE TITLE
[question]: add mocha xqsuite script from generator

### ISSUE THREAD

**Issue by duncdrum** [eXist-db/node-exist#124]
I now have a pretty robust and dynamic test generator for xqsuite tests executed on the server and then "parsed" by Mocha. see [generators/app/templates/tests/xqs/xqSuite.js](https://github.com/eXist-db/generator-exist/blob/master/generators/app/templates/tests/xqs/xqSuite.js). Should we include that here? 

The only thing that we would need to adjust is adding a method to provide the location of the xqsuite runner (and or suite) file. 

The downside it would add Mocha as a dependency. 

The upside you could run xqsuite tests from inside node-exist. 

@line-o what do you think?

**Comment by line-o** [comment]
I do like your approach - it is also very similar to https://github.com/line-o/xbow/blob/master/test/mocha/xqSuite.js. This would add mocha as a dependency, too.
I do like your approach to directly run the test-runner instead of uploading it. My version does test more metrics though. I think we should pick the best of both.

**Comment by duncdrum** [comment]
Yes great minds thinking alike … i have these three [fixtures](https://github.com/duncdrum/generator-exist/commit/165a12b02bd2422c455e8c207ce4dd53f6779c9a) of different junit-reports i encountered (and slimmed down) in the wild. Most of my work was making sure that all three kind of runs are reported properly, and that repeated runs stay consistent. 

How does your code deal with repeated runs, and with those variations in report format?

**Comment by line-o** [comment]
I would like to pick this up after a while of hibernation. How about a paired programming session to get this included?
What I would like to achieve:
- the level of reporting your script returns which looks much better than what I currently have
- do that with tape instead of mocha to not add a new dependency

How does that sound @duncdrum?
